### DISCUSSION TITLE
Incress ranges

### DISCUSSION THREAD

**Discussion by KKarmugil** [dchristl/esp32_nat_router_extended#50] [Ideas]
By reducing or increasing package size can wifi range be increased ?  If its possible can you add selecting package size because i need range even speed is low then 100kb also ok for me i am trying share wifi for long distance with my friend for whatsapp

**Comment by dchristl** [comment]
Hello KKarmugli,

sorry that is not how this works. The microcontroller is just a gateway and can't change the packet size. The only possible solution for long range is called LoRaWAN. This is a special mode and only usable for some sensors submitting only Bytes, because this decreases the speed below 10 kB/s. But how about to use multiple esp32 over the distance? 

Regards,
Danny

**Reply by KKarmugil** [reply]
thank you dchristl

**Comment by pipi61** [comment]
Hi!
use an external antenna to increase the range. 
This work: https://www.hackster.io/simon-vavpotic/esp32-and-esp8266-external-antenna-f28e6b
I tried to cut the wire and solder the antenna of an old router

**Comment by Verpeilimon** [comment]
Hi Danny,
A few weeks ago, I came across your really cool ESP32 NAT ROUTER. The "repeating" works absolutely great and the graphical interface not only offers many possibilities but is also so simple that even laypeople should be able to handle it.
Now you are probably wondering why I am writing a post here if I am so impressed with the project and it works so brilliantly for me. To better understand, I would like to tell you a short story... During the two "peak years" of Corona, I bought a set of laser tag guns for myself and my family and researched for a long time which ones would be best for us. The decision was made for this system: https://battlecompany.com/product/brx-laser-tag-rifle/ The "taggers" are not normally centrally controllable and do not really offer scoring. I came across an open source project that makes this possible with the following setup:
Hosting:
1 ESP32 (web server, wired via IO pins to transceiver)
1 ESP32 (transceiver, wired via IO pins to web server)
Client:
1 ESP32 (connects by Bluetooth to tagger and creates an ESPnow network with all other taggers and the host (web server+transceiver)
It works great outside and also with a large range. As soon as walls are in between, the connection quickly breaks off and the taggers may leave the game.
Now I wanted to use the ESP32 NAT Router project to build several repeaters that log into the taggers' WIFI and expand the existing network. Unfortunately, there are always interruptions. And as soon as two routers/repeaters are needed to reach the server, everything breaks down or the network is occasionally simply "GONE".
Would it be possible to create a connection via Lora with your project? So that one router connects to the tagger client per tagger and passes it on to all others via Lora?
I would be willing to make a generous donation for a solution, especially if it involves Lora, because I would really like to be able to play "freely/outside" 😊the sneded and received information is really minimalistic, so i think it could work ?
Best regards,
Daniel

**Comment by dchristl** [comment]
Hi @Verpeilimon ,

thank you for your kind words and for your interesting Laser Tag project. What I haven't quite understood is where my project fits into your setup? I hope, I have understtod everything. In principle, what you're planning is technically feasible, but it's not quite so straightforward. Even if my project could 'speak' LoRa, all the other devices would need to as well. Unlike in Wi-Fi, where, for example, Ethernet is used, here, the LoRaWAN protocol is used. Therefore, the client (connected via Bluetooth) would also need to speak LoRaWAN for it to work. Is this the case, otherwise adjustments would need to be made. What would also be necessary is a 'translation' between ESP-NOW and LoRaWAN since these are also different protocols.

You see, it is not so easy as it seems. Unfortunately, in the ESP32 NAT router, I don't see this additional protocol because it's more focused on high data rates rather than power savings or very long-range communication.

I think it would be better to address your plan in a separate project and implement communication specifically for your use case. In that case, I would suggest sticking with ESP-NOW and possibly adding a few additional nodes (like Mesh). In theory, you would only need to distribute a few additional ESP32 devices with ESP-Now in your rooms to ensure better coverage. I could even imagine that your project might already be capable of this (e.g., ESP32 clients of the tagger without weapons?). I'm not familiar enough with ESP32-Now, but from what I've read, it's possible to operate it in a mesh-like manner. I can't assess whether this is implemented in your project. Do you have a link for reference?

Kind regards,
Danny

**Comment by Verpeilimon** [comment]
Hi Danny,

First of all i will send you the links to thelatest vresion(s) in use with some describing words:

This ist he version i`m using and is (nearly) running without failures, and also the one i`m using:

Webserver,Transreceiver,Tagger – Files are included into this file: https://ahlmann-kleve.de/ESP32ForTransceiver-10-9-A.zip

The system is running a network from each tagger to each other and tot he webserver.
That seems tob e the beast possibility but also the biggest limitation i think.

The hope of mine was / ist hat it would be possible to creat a second network „lorawan“ that is connected be a couple of an rylr896 modulfor the communication of the new „lorawan“ and an esp32, translating the existing network  and exchanges to ech tagger and the server the neccessary information oft he game.

I think that would be the way to get a more stable network with much bigger range.
The latest try was a box with an esp32 wroom and an powerbank as powersupply, an on / off switch and a port to recharge.
But the connection worksmost just a time and disappears immediatly 8 think cause oft he range and connection oft he tagger) especially if there are more than a single „box“ / „repeater“ 
It is not easy to explain but i hope it would be possible to understand 😊.
Best wishes

Daniel

**Comment by dchristl** [comment]
Hi Daniel, 

> Webserver,Transreceiver,Tagger – Files are included into this file: https://ahlmann-kleve.de/ESP32ForTransceiver-10-9-A.zip 

These are only the binaries. Do you have the sources? 

> The system is running a network from each tagger to each other and tot he webserver.
That seems tob e the beast possibility but also the biggest limitation i think.
The hope of mine was / ist hat it would be possible to creat a second network „lorawan“ that is connected be a couple of an rylr896 modulfor the communication of the new „lorawan“ and an esp32, translating the existing network and exchanges to ech tagger and the server the neccessary information oft he game.

The network is ESP-now, right? This is not compatible to "normal" networks or do you already managed to put the ESP32-NAT-Router inside this network? If so, can you describe the topology? At which point is internet (no ESP-now) involved?

In my opinion, since adjustments to the Tagger firmware need to be made anyway, it seems most sensible to simply expand the ESP-now network with additional nodes that increase the range. These nodes would be located in specific places in the rooms, not on the Tagger itself. They would essentially act as additional Taggers, but without actually tagging; their sole purpose would be to extend the network. The rationale behind this is that the Taggers on the weapons already communicate with each other and establish the network. I believe this could be implemented quite easily, but I'm still not entirely sure if I've fully understood everything correctly. ;)

**Comment by Verpeilimon** [comment]
Hi Danny,

In the past few days, I have been trying to get the source files for the binaries, but unfortunately, the project and the community have been a bit "dormant" in the meantime.

I am afraid that it will not be possible for me to get these files so quickly.

In the past few days, I have also experimented a bit and tried to visualize everything.

It actually works to activate ONE ROUTER / REPEATER, turn on the taggers, then the web server, and I can configure the game outside the actual range of the web server / transceiver, or use the router as a "bridge" to bridge e.g. thick walls.

It does not work if I use multiple of these routers / repeaters, then it seems as if the tagggers are not able to decide with whom to connect.

I thought that there might be a way to carry an ESP32 router directly with each tagger that connects to this esp32 immediately after turning it on.

However, these additional routers / repeaters then connect to LoraWan with each other and the websverer / transceiver, ... the question is whether that would even work.

Under this link, I have tried to describe it to you once:

https://ahlmann-kleve.de/Toolz/jedgerepeat.pdf

What do you think?

All the best

Daniel

**Comment by dchristl** [comment]
Hello Verpeilimon,

Thank you for your explanations, but I haven't fully grasped it yet. Maybe I'm not the best person to discuss this (I'm more of a developer than a network expert). From my perspective, everything looks extremely complicated, and it seems like there would be a need for implementations at many points. For example, if the taggers communicate via ESP-Now, there would always need to be a compatibility layer (ESP-NOW <-> LoRa), or else the connection wouldn't work. LoRa is also quite slow, so I'm not sure if the data rate would be sufficient. It wouldn't be useful if a hit arrives five minutes later.

Where I'm struggling the most mentally is why this should be necessary in the first place. All ESP-Now devices (the taggers) build a network among themselves and push the data to the web server (via which protocol, actually?), depending on which tagger is available/nearest at the moment. Multiple hops are used to ultimately reach the web server. This works just like a mesh, if I understand the documentation correctly. That's why I had the idea of adding additional taggers that only exist in the rooms to make the network denser/stronger. In my opinion, this should be sufficient; it would work in a mesh network. No additional ESP or other router would be required. It would be relatively simple and feasible without any modifications to the code. The only downside would be that you would need a few additional taggers, but that should be manageable.

Kind regards,
Danny
